providers/aws/kms.go

```diff
@@ -0,0 +1,97 @@
+package aws
+
+import (
+	"github.com/Qovery/pleco/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"strconv"
+)
+
+func getKeys(svc kms.KMS, tagName string) []*kms.KeyListEntry{
+	input := &kms.ListKeysInput{}
+	keys, err := svc.ListKeys(input)
+	handleKMSError(err)
+
+	return keys.Keys
+}
+
+func deleteKey(svc kms.KMS,keyId *string) (*kms.ScheduleKeyDeletionOutput,error){
+	input := &kms.ScheduleKeyDeletionInput{
+		KeyId:               keyId,
+		PendingWindowInDays: aws.Int64(7),
+	}
+	result, err := svc.ScheduleKeyDeletion(input)
+	handleKMSError(err)
+
+	return result,err
+}
+
+func getKeyTags (svc kms.KMS, keyId *string) []*kms.Tag {
+	input := &kms.ListResourceTagsInput{
+		KeyId: keyId,
+	}
+
+	tags, err := svc.ListResourceTags(input)
+	handleKMSError(err)
+
+	return tags.Tags
+}
+
+func getKeyMetadata (svc kms.KMS,keyId *string) *kms.DescribeKeyOutput{
+	input := &kms.DescribeKeyInput{KeyId: keyId}
+	data, err := svc.DescribeKey(input)
+	handleKMSError(err)
+
+	return data
+}
+
+func handleKMSError (error error) {
+	if error != nil {
+		if aerr, ok := error.(awserr.Error); ok {
+			switch aerr.Code() {
+			case kms.ErrCodeNotFoundException:
+				println(kms.ErrCodeNotFoundException, aerr.Error())
+			case kms.ErrCodeInvalidArnException:
+				println(kms.ErrCodeInvalidArnException, aerr.Error())
+			case kms.ErrCodeDependencyTimeoutException:
+				println(kms.ErrCodeDependencyTimeoutException, aerr.Error())
+			case kms.ErrCodeInternalException:
+				println(kms.ErrCodeInternalException, aerr.Error())
+			case kms.ErrCodeInvalidStateException:
+				println(kms.ErrCodeInvalidStateException, aerr.Error())
+			default:
+				println(aerr.Error())
+			}
+		} else {
+			println(error.Error())
+		}
+
+	}
+}
+
+func deleteExpiredKeys(svc kms.KMS, tagName string, dryRun bool) error{
+	keys := getKeys(svc, tagName)
+	var numberOfKeysToDelete int64
+	for _, key :=range keys {
+		tags:= getKeyTags(svc,key.KeyId)
```

Comment:
linter is not enabled yet, but try to be consistent (you should be able to do it from your IDE):
x:= y
x :=y
x := y